text/0000-deprecate-globals-resolver.md

```diff
@@ -0,0 +1,122 @@
+- Start Date: 2018-05-08
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Deprecate all use of:
+
+- Ember Globals Resolver (looks up a class via a global namespace such as "App")
+- Creation of a Global Namespace (`var App = Ember.Namespace.create();`)
+- Ember.TEMPLATES array
+- &lt;script type="text/handlebars" data-template-name="path/to/template"&gt;
+
+Use of any of the above should trigger a deprecation warning, with a target
+of version 4.0
+
+# Motivation
+
+Over the past years we have transitioned to using Ember-CLI as the main way
+to compile Ember apps. The globals resolver is a holdover and primarily
+facilitates use of Ember without Ember-CLI.
```

Comment:
I did a quick review, and it does look like the [ember-cli classic resolver](https://github.com/ember-cli/ember-resolver/blob/master/addon/resolvers/classic/index.js) replaces all methods on the [globals resolver](https://github.com/emberjs/ember.js/blob/master/packages/@ember/application/globals-resolver.js) 👍